domain/src/main/kotlin/no/nav/su/se/bakover/domain/Behandling.kt

```diff
@@ -3,6 +3,9 @@ package no.nav.su.se.bakover.domain
 import arrow.core.Either
 import arrow.core.left
 import arrow.core.right
+import io.micrometer.core.instrument.Counter
```

Comment:
Er det ikke litt krise at domenet vårt skal være avhenging av dette? Burde innføre en abstraksjon som snur om på denne avhengigheten.. Dette gjelder forsåvidt det som allerede er lagt inn for service også.

Reply:
Selve konseptet metrikk(er) vil jeg si er at domeneanliggende. På samme måte som logging er det. Og forsåvidt persistering. Så kan man jo så klart lage en abstraksjon (som må vedlikeholdes). Jeg ser litt på micrometer på samme måte som for slf4j. Hvis du har laget et logging/metrikk-lib og ikke støtter slf4j/micrometer er du ganske useriøs. Men det er jo min mening.

Fred George og gjengen ville garantert kjørt Observer style på begge disse konseptene. Men jeg mener fremdeles disse observerne burde ligge i domene-laget, siden domenet skal bestemme hva og når noe skal logges/metrikkes hvis det er domene-logikk anliggender som f.eks. ny søknad og behandlingsting.

Reply:
For meg handler ikke dette om hvilket design-pattern det er implementert med. Det handler om at domenet vårt ikke bør kjenne til detaljene rundt hvordan denne tellingen utføres - være seg micrometer eller mysupercountinglib.

Reply:
Hmm, jeg skal tenke litt på den over helga. Vanligvis ville jeg vel lagt infrastruktur-biten av dette under behandling/infrastructure/metrics og søknad/infrastructure/metrics. Men vi har ikke noe generelt begrep for infrastructure atm, så da blir det evt. en ny modul, eller stappe det i web eller service. Ville uansett hatt et interface per domeneområde i domain/service (hvor jeg vanligvis bruker samlebegrepet application), litt ala. Persistence-laget.